code/src/test/kotlin/lexicalTests/LexerTest.kt

```diff
@@ -76,6 +74,176 @@ class LexerTest {
+    }
+
+    @org.junit.jupiter.api.Disabled
+    @org.junit.jupiter.api.Test
+    fun lexerTestIllegalIdentifiers() {
+        val lex = Lexer("var = \"id \"id\" 'di 'di' [] { () true \\ , }")
+        //Tokens: var, =, "id, "id", 'di, 'di', [, ], {, (, ), true, \, ",", ->, }
+        val tokens = lex.getTokenSequence().toList()
+
+        tokens.dropLast(1).forEach{ assertTrue(it.token !is lexer.Token.Identifier) }
+        assertTrue(tokens.last().token is lexer.Token.SpecialChar.EndOfLine)
+    }
+
+    @org.junit.jupiter.api.Test
+    fun lexerTestBooleanLiteralTokens() {
+        val lex = Lexer("true false")
+
+        val tokens = lex.getTokenSequence().toList()
+
+        assertTrue(tokens[0].token is lexer.Token.Literal.Bool)
+        assertTrue(tokens[1].token is lexer.Token.Literal.Bool)
+    }
+
+    @org.junit.jupiter.api.Test
+    fun lexerTestFunctionDeclaration() {
+        val lex = Lexer("func print = (txt message) : none\n{ }")
+
+        val tokens = lex.getTokenSequence().toList()
+
+        assertPositionalToken(tokens[0],
+                { token -> token is lexer.Token.Type.Func },
+                0, 0)
+        assertPositionalToken(tokens[1],
+                { token -> token is lexer.Token.Identifier && token.value == "print" },
+                5, 0)
+        assertPositionalToken(tokens[2],
+                { token -> token is lexer.Token.SpecialChar.Equals },
+                11, 0)
+        assertPositionalToken(tokens[3],
+                { token -> token is lexer.Token.SpecialChar.ParenthesesStart },
+                13, 0)
+        assertPositionalToken(tokens[4],
+                { token -> token is lexer.Token.Type.Text },
+                14, 0)
+        assertPositionalToken(tokens[5],
+                { token -> token is lexer.Token.Identifier && token.value == "message" },
+                18, 0)
+        assertPositionalToken(tokens[6],
+                { token -> token is lexer.Token.SpecialChar.ParenthesesEnd },
+                25, 0)
+        assertPositionalToken(tokens[7],
+                { token -> token is lexer.Token.SpecialChar.Colon },
+                27, 0)
+        assertPositionalToken(tokens[8],
+                { token -> token is lexer.Token.Type.None },
+                29, 0)
+        assertPositionalToken(tokens[9],
+                { token -> token is lexer.Token.SpecialChar.EndOfLine },
+                33, 0)
+        assertPositionalToken(tokens[10],
+                { token -> token is lexer.Token.SpecialChar.BlockStart },
+                0, 1)
+        assertPositionalToken(tokens[11],
+                { token -> token is lexer.Token.SpecialChar.BlockEnd },
+                2, 1)
+        assertPositionalToken(tokens[12],
+                { token -> token is lexer.Token.SpecialChar.EndOfLine },
+                3, 1)
+    }
+
+    @org.junit.jupiter.api.Test
+    fun lexerTestDoesntRequireWhitespaces() {
+        val lex = Lexer("x=y[0]")
```

Comment:
test this with types followed by special char as well (as previously commented..) something like "(txt" and "=(" - both cases needs testing